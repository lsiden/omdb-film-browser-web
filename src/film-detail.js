import React from "react"
import PropTypes from "prop-types"
import { connect } from "react-redux"
import Spinner from "respin"

import { CloseButton } from "components/close-button"
import { ESC_KEY } from "./constants"
import { scrollToTop } from "components/scroll"
import ButtonLink from "components/button-link"
import { fetchFilmDetails } from "actions/remote"
import { headerStyle as bannerHeaderStyle } from "style"
import { BANNER_TITLE, FETCH_TIMEOUT } from "./constants"

const titleStyle = {
  marginBottom: "5pt",
}
const detailsStyle = {
  marginTop: 20,
  fontSize: "14pt",
}
const topButtonStyle = {
  marginBottom: 20,
}
const spinnerWrapperStyle = {
  height: "100vh",
  minHeight: "100vh",
  display: "flex",
  justifyContent: "center",
}

const itemExists = (item) => item && item !== "N/A"

const imdbUrl = (imdbID) => `https://www.imdb.com/title/${imdbID}`

const titleBannerStyle = {
  ...bannerHeaderStyle,
  display: "flex",
  justifyContent: "space-between",
}

const TitleBanner = ({ title }) => (
  <header style={titleBannerStyle}>
    <div style={{ lineHeight: 1 }}>
      <h1 style={{ fontSize: 24 }}>{BANNER_TITLE}</h1>
      <br />
      <h1 style={{ ...titleStyle, marginTop: 0 }}>{title}</h1>
    </div>
    <CloseButton buttonStyle={{ color: "white" }} />
  </header>
)
TitleBanner.propTypes = {
  title: PropTypes.string.isRequired,
}

export class FilmDetail extends React.Component {
  constructor(props) {
    const { imdbID } = props
    super(props)
    this.keyDownListener = this.keyDownListener.bind(this)
    this.imdbUrl = `https://www.imdb.com/title/${imdbID}`
  }

  keyDownListener(ev) {
    if (ev.keyCode === ESC_KEY) {
      window.location.href = "/"
    }
  }

  componentDidMount() {
    document.addEventListener("keydown", this.keyDownListener)
    setTimeout(() => {
      if (!this.props.filmDetails) {
        window.location.href = "/"
      } else {
        const { dispatchFetchFilmDetails, imdbID } = this.props
        dispatchFetchFilmDetails(imdbID)
      }
    }, FETCH_TIMEOUT)
  }

  componentWillUnmount() {
    document.removeEventListener("keydown", this.keyDownListener)
  }

  renderTitle() {
    const { filmDetails } = this.props
    return <TitleBanner title={filmDetails.Title} />
  }

  renderDetails() {
    const { filmDetails } = this.props
    return (
      <ul style={detailsStyle}>
        <li>{filmDetails.Year}</li>
        <li>Directed by {filmDetails.Director}</li>
        <li>Written by {filmDetails.Writer}</li>
        <li>Cast: {filmDetails.Actors}</li>
        <li>Language: {filmDetails.Language}</li>
        {itemExists(filmDetails.Awards) && (
          <li>Awards: {filmDetails.Awards}</li>
        )}
        <li>Run Time: {filmDetails.Runtime}</li>
        <li>IMDB Rating: {filmDetails.imdbRating}/10</li>
        <li>Box Office: {filmDetails.BoxOffice}</li>
        {itemExists(filmDetails.Website) && (
          <li>
            <a href={filmDetails.Website}>Official website</a>
          </li>
        )}
        <li>
          <a href={imdbUrl(filmDetails.imdbID)}>{"IMDB page"}</a>
        </li>
      </ul>
    )
  }

  renderScrollToTopButton() {
    return (
      <ButtonLink
        style={topButtonStyle}
        onClick={scrollToTop}
        title="Scroll to top of page"
      >
        Top
      </ButtonLink>
    )
  }

  render() {
    const { filmDetails, isFetching } = this.props
    return isFetching ? (
      <div style={spinnerWrapperStyle}>
        <Spinner size={64} />
      </div>
    ) : filmDetails ? (
      <React.Fragment>
        {this.renderTitle()}
        <div style={{ marginLeft: 20 }}>
          <img
            src={filmDetails.Poster}
            alt="poster"
            style={{ marginTop: 10 }}
          />
          {this.renderDetails()}
          {this.renderScrollToTopButton()}
        </div>
      </React.Fragment>
    ) : null
  }
}

FilmDetail.propTypes = {
  imdbID: PropTypes.string.isRequired,
  filmDetails: PropTypes.object,
  isFetching: PropTypes.bool.isRequired,
  dispatchFetchFilmDetails: PropTypes.func.isRequired,
}

const ConnectedFilmDetail = connect(
  (state) => ({
    filmDetails: state.filmDetails,
    isFetching: !!state.isFetching,
  }),
  (dispatch) => ({
    dispatchFetchFilmDetails: (imdbID) => dispatch(fetchFilmDetails(imdbID)),
  })
)(FilmDetail)

const RoutedFilmDetail = ({ match }) => {
  const { imdbID } = match.params
  return <ConnectedFilmDetail imdbID={imdbID} />
}

RoutedFilmDetail.propTypes = {
  match: PropTypes.object.isRequired,
}

export default RoutedFilmDetail
