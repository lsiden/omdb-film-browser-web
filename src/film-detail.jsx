import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import Spinner from 'respin'

import CloseButton from 'components/close-button'
import NavButton from 'components/nav-button'
import { scrollToTop } from 'components/scroll'
import { fetchFilmDetails } from 'actions/remote'
import { headerStyle as bannerHeaderStyle } from 'style'
import { ESC_KEY, BANNER_TITLE, FETCH_TIMEOUT } from './constants'

const titleStyle = {
  marginBottom: '5pt',
}
const detailsStyle = {
  marginTop: 20,
  fontSize: '14pt',
}
const topButtonStyle = {
  marginBottom: 20,
}
const spinnerWrapperStyle = {
  height: '100vh',
  minHeight: '100vh',
  display: 'flex',
  justifyContent: 'center',
}

const itemExists = (item) => item && item !== 'N/A'
const imdbUrl = (imdbID) => `https://www.imdb.com/title/${imdbID}`

const titleBannerStyle = {
  ...bannerHeaderStyle,
  display: 'flex',
  justifyContent: 'space-between',
}

const TitleBanner = ({ title }) => (
  <header style={titleBannerStyle}>
    <div style={{ lineHeight: 1 }}>
      <h1 style={{ fontSize: 24 }}>{BANNER_TITLE}</h1>
      <br />
      <h1 style={{ ...titleStyle, marginTop: 0 }}>{title}</h1>
    </div>
    <CloseButton buttonStyle={{ color: 'white' }} />
  </header>
)
TitleBanner.propTypes = {
  title: PropTypes.string.isRequired,
}

export class FilmDetail extends React.Component {
  static keyDownListener(ev) {
    if (ev.keyCode === ESC_KEY) {
      window.location.href = '/'
    }
  }

  constructor(props) {
    const { imdbID } = props
    super(props)
    this.imdbUrl = `https://www.imdb.com/title/${imdbID}`
  }

  componentDidMount() {
    document.addEventListener('keydown', FilmDetail.keyDownListener)
    setTimeout(() => {
      if (!this.props.filmDetails) {
        window.location.href = '/'
      } else {
        const { dispatchFetchFilmDetails, imdbID } = this.props
        dispatchFetchFilmDetails(imdbID)
      }
    }, FETCH_TIMEOUT)
  }

  componentWillUnmount() {
    document.removeEventListener('keydown', FilmDetail.keyDownListener)
  }

  renderTitle() {
    const { filmDetails } = this.props
    return <TitleBanner title={filmDetails.Title} />
  }

  renderDetails() {
    const { filmDetails } = this.props
    return (
      <ul style={detailsStyle}>
        <li>{filmDetails.Year}</li>
        <li>
          Directed by
          {filmDetails.Director}
        </li>
        <li>
          Written by
          {filmDetails.Writer}
        </li>
        <li>
          Cast:
          {filmDetails.Actors}
        </li>
        <li>
          Language:
          {filmDetails.Language}
        </li>
        {itemExists(filmDetails.Awards) && (
          <li>
            Awards:
            {filmDetails.Awards}
          </li>
        )}
        <li>
          Run Time:
          {filmDetails.Runtime}
        </li>
        <li>
          IMDB Rating:
          {filmDetails.imdbRating}
          /10
        </li>
        <li>
          Box Office:
          {filmDetails.BoxOffice}
        </li>
        {itemExists(filmDetails.Website) && (
          <li>
            <a href={filmDetails.Website}>Official website</a>
          </li>
        )}
        <li>
          <a href={imdbUrl(filmDetails.imdbID)}>IMDB page</a>
        </li>
      </ul>
    )
  }

  static renderScrollToTopButton() {
    return (
      <NavButton
        style={topButtonStyle}
        onClick={scrollToTop}
        title="Scroll to top of page"
      >
        Top
      </NavButton>
    )
  }

  render() {
    const { filmDetails, isFetching } = this.props
    if (isFetching) {
      return (
        <div style={spinnerWrapperStyle}>
          <Spinner size={64} />
        </div>
      )
    }

    if (filmDetails) {
      return (
        <div>
          {this.renderTitle()}
          <div style={{ marginLeft: 20 }}>
            <img
              src={filmDetails.Poster}
              alt="poster"
              style={{ marginTop: 10 }}
            />
            {this.renderDetails()}
            {FilmDetail.renderScrollToTopButton()}
          </div>
        </div>
      )
    }
    return null
  }
}

FilmDetail.propTypes = {
  imdbID: PropTypes.string.isRequired,
  filmDetails: PropTypes.object.isRequired,
  isFetching: PropTypes.bool.isRequired,
  dispatchFetchFilmDetails: PropTypes.func.isRequired,
}

const ConnectedFilmDetail = connect(
  (state) => ({
    filmDetails: state.filmDetails,
    isFetching: !!state.isFetching,
  }),
  (dispatch) => ({
    dispatchFetchFilmDetails: (imdbID) => dispatch(fetchFilmDetails(imdbID)),
  }),
)(FilmDetail)

const RoutedFilmDetail = ({ match }) => {
  const {
    imdbID, dispatchFetchFilmDetails,
  } = match.params
  return (
    <ConnectedFilmDetail
      imdbID={imdbID}
      dispatchFetchFilmDetails={dispatchFetchFilmDetails}
    />
  )
}

RoutedFilmDetail.propTypes = {
  match: PropTypes.object.isRequired,
}

export default RoutedFilmDetail
