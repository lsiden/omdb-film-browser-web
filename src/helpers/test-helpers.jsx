import React from 'react'
import { Provider } from 'react-redux'
import { StaticRouter } from 'react-router-dom'

/* eslint-disable-next-line import/no-extraneous-dependencies */
import { create } from 'react-test-renderer'

import store from 'store'

export const createWithContext = (ui) => create(
  <Provider store={store}>
    <StaticRouter context={{}}>
      {ui}
    </StaticRouter>
  </Provider>
)
