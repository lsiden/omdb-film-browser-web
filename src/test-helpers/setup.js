import { configure } from "enzyme"
import Adapter from "enzyme-adapter-react-16"

configure({ adapter: new Adapter() })

jest.mock("abort-controller", () => class {})
jest.mock("event-target-shim", () => {})
