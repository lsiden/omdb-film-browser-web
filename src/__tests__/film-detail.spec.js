import React from 'react'
import { shallow } from 'enzyme'

import 'test-helpers/setup'
import {FilmDetail} from 'film-detail'

const films = require('./films.json').Search
let  dispatchViewList

const defaultProps = () => {
  dispatchViewList = jest.fn()
  return {
    filmSummary: films[0],
    dispatchViewList,
  }
}

const createWrapper = (props={}) => shallow(
  <FilmDetail {...{...defaultProps(), ...props}} />
)

it('renders a title', () => {
  const wrapper = createWrapper()
  expect(wrapper.text()).toEqual(expect.stringContaining(films[0].Title))
})

test('click on close invokes dispatchViewList()', () => {
  const wrapper = createWrapper()
  wrapper.find('#close-button').simulate('click')
  expect(dispatchViewList).toHaveBeenCalled()
})
