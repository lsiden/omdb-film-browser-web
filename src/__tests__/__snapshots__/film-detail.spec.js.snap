// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilmDetail
    dispatchFetchFilmDetails={[Function]}
    filmDetails={
        Object {
            "Poster": "https://images-na.ssl-images-amazon.com/images/M/MV5BODAzMDgxMDc1MF5BMl5BanBnXkFtZTgwMTI0OTAzMjE@._V1_SX300.jpg",
            "Title": "The Hobbit: The Battle of the Five Armies",
            "Type": "movie",
            "Year": "2014",
            "imdbID": "tt2310332",
          }
    }
    imdbID="tt2310332"
    isFetching={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <TitleBanner
          title="The Hobbit: The Battle of the Five Armies"
/>,
        <div
          style={
                    Object {
                              "marginLeft": 20,
                            }
          }
>
          <img
                    alt="poster"
                    src="https://images-na.ssl-images-amazon.com/images/M/MV5BODAzMDgxMDc1MF5BMl5BanBnXkFtZTgwMTI0OTAzMjE@._V1_SX300.jpg"
                    style={
                              Object {
                                        "marginTop": 10,
                                      }
                    }
          />
          <ul
                    style={
                              Object {
                                        "fontSize": "14pt",
                                        "marginTop": 20,
                                      }
                    }
          >
                    <li>
                              2014
                    </li>
                    <li>
                              Directed by 
                    </li>
                    <li>
                              Written by 
                    </li>
                    <li>
                              Cast: 
                    </li>
                    <li>
                              Language: 
                    </li>
                    <li>
                              Run Time: 
                    </li>
                    <li>
                              IMDB Rating: 
                              /10
                    </li>
                    <li>
                              Box Office: 
                    </li>
                    <li>
                              <a
                                        href="https://www.imdb.com/title/tt2310332"
                              >
                                        IMDB page
                              </a>
                    </li>
          </ul>
          <ButtonLink
                    onClick={[Function]}
                    style={
                              Object {
                                        "marginBottom": 20,
                                      }
                    }
                    title="Scroll to top of page"
          >
                    Top
          </ButtonLink>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "title": "The Hobbit: The Battle of the Five Armies",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="poster"
              src="https://images-na.ssl-images-amazon.com/images/M/MV5BODAzMDgxMDc1MF5BMl5BanBnXkFtZTgwMTI0OTAzMjE@._V1_SX300.jpg"
              style={
                            Object {
                                          "marginTop": 10,
                                        }
              }
/>,
            <ul
              style={
                            Object {
                                          "fontSize": "14pt",
                                          "marginTop": 20,
                                        }
              }
>
              <li>
                            2014
              </li>
              <li>
                            Directed by 
              </li>
              <li>
                            Written by 
              </li>
              <li>
                            Cast: 
              </li>
              <li>
                            Language: 
              </li>
              <li>
                            Run Time: 
              </li>
              <li>
                            IMDB Rating: 
                            /10
              </li>
              <li>
                            Box Office: 
              </li>
              <li>
                            <a
                                          href="https://www.imdb.com/title/tt2310332"
                            >
                                          IMDB page
                            </a>
              </li>
</ul>,
            <ButtonLink
              onClick={[Function]}
              style={
                            Object {
                                          "marginBottom": 20,
                                        }
              }
              title="Scroll to top of page"
>
              Top
</ButtonLink>,
          ],
          "style": Object {
            "marginLeft": 20,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "poster",
              "src": "https://images-na.ssl-images-amazon.com/images/M/MV5BODAzMDgxMDc1MF5BMl5BanBnXkFtZTgwMTI0OTAzMjE@._V1_SX300.jpg",
              "style": Object {
                "marginTop": 10,
              },
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  2014
</li>,
                <li>
                  Directed by 
</li>,
                <li>
                  Written by 
</li>,
                <li>
                  Cast: 
</li>,
                <li>
                  Language: 
</li>,
                undefined,
                <li>
                  Run Time: 
</li>,
                <li>
                  IMDB Rating: 
                  /10
</li>,
                <li>
                  Box Office: 
</li>,
                undefined,
                <li>
                  <a
                                    href="https://www.imdb.com/title/tt2310332"
                  >
                                    IMDB page
                  </a>
</li>,
              ],
              "style": Object {
                "fontSize": "14pt",
                "marginTop": 20,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "2014",
                },
                "ref": null,
                "rendered": "2014",
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Directed by ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Directed by ",
                  undefined,
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Written by ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Written by ",
                  undefined,
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Cast: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Cast: ",
                  undefined,
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Language: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Language: ",
                  undefined,
                ],
                "type": "li",
              },
              undefined,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Run Time: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Run Time: ",
                  undefined,
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "IMDB Rating: ",
                    undefined,
                    "/10",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "IMDB Rating: ",
                  undefined,
                  "/10",
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Box Office: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Box Office: ",
                  undefined,
                ],
                "type": "li",
              },
              undefined,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    href="https://www.imdb.com/title/tt2310332"
>
                    IMDB page
</a>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "IMDB page",
                    "href": "https://www.imdb.com/title/tt2310332",
                  },
                  "ref": null,
                  "rendered": "IMDB page",
                  "type": "a",
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Top",
              "onClick": [Function],
              "style": Object {
                "marginBottom": 20,
              },
              "title": "Scroll to top of page",
            },
            "ref": null,
            "rendered": "Top",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <TitleBanner
            title="The Hobbit: The Battle of the Five Armies"
/>,
          <div
            style={
                        Object {
                                    "marginLeft": 20,
                                  }
            }
>
            <img
                        alt="poster"
                        src="https://images-na.ssl-images-amazon.com/images/M/MV5BODAzMDgxMDc1MF5BMl5BanBnXkFtZTgwMTI0OTAzMjE@._V1_SX300.jpg"
                        style={
                                    Object {
                                                "marginTop": 10,
                                              }
                        }
            />
            <ul
                        style={
                                    Object {
                                                "fontSize": "14pt",
                                                "marginTop": 20,
                                              }
                        }
            >
                        <li>
                                    2014
                        </li>
                        <li>
                                    Directed by 
                        </li>
                        <li>
                                    Written by 
                        </li>
                        <li>
                                    Cast: 
                        </li>
                        <li>
                                    Language: 
                        </li>
                        <li>
                                    Run Time: 
                        </li>
                        <li>
                                    IMDB Rating: 
                                    /10
                        </li>
                        <li>
                                    Box Office: 
                        </li>
                        <li>
                                    <a
                                                href="https://www.imdb.com/title/tt2310332"
                                    >
                                                IMDB page
                                    </a>
                        </li>
            </ul>
            <ButtonLink
                        onClick={[Function]}
                        style={
                                    Object {
                                                "marginBottom": 20,
                                              }
                        }
                        title="Scroll to top of page"
            >
                        Top
            </ButtonLink>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "title": "The Hobbit: The Battle of the Five Armies",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="poster"
                src="https://images-na.ssl-images-amazon.com/images/M/MV5BODAzMDgxMDc1MF5BMl5BanBnXkFtZTgwMTI0OTAzMjE@._V1_SX300.jpg"
                style={
                                Object {
                                                "marginTop": 10,
                                              }
                }
/>,
              <ul
                style={
                                Object {
                                                "fontSize": "14pt",
                                                "marginTop": 20,
                                              }
                }
>
                <li>
                                2014
                </li>
                <li>
                                Directed by 
                </li>
                <li>
                                Written by 
                </li>
                <li>
                                Cast: 
                </li>
                <li>
                                Language: 
                </li>
                <li>
                                Run Time: 
                </li>
                <li>
                                IMDB Rating: 
                                /10
                </li>
                <li>
                                Box Office: 
                </li>
                <li>
                                <a
                                                href="https://www.imdb.com/title/tt2310332"
                                >
                                                IMDB page
                                </a>
                </li>
</ul>,
              <ButtonLink
                onClick={[Function]}
                style={
                                Object {
                                                "marginBottom": 20,
                                              }
                }
                title="Scroll to top of page"
>
                Top
</ButtonLink>,
            ],
            "style": Object {
              "marginLeft": 20,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "poster",
                "src": "https://images-na.ssl-images-amazon.com/images/M/MV5BODAzMDgxMDc1MF5BMl5BanBnXkFtZTgwMTI0OTAzMjE@._V1_SX300.jpg",
                "style": Object {
                  "marginTop": 10,
                },
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li>
                    2014
</li>,
                  <li>
                    Directed by 
</li>,
                  <li>
                    Written by 
</li>,
                  <li>
                    Cast: 
</li>,
                  <li>
                    Language: 
</li>,
                  undefined,
                  <li>
                    Run Time: 
</li>,
                  <li>
                    IMDB Rating: 
                    /10
</li>,
                  <li>
                    Box Office: 
</li>,
                  undefined,
                  <li>
                    <a
                                        href="https://www.imdb.com/title/tt2310332"
                    >
                                        IMDB page
                    </a>
</li>,
                ],
                "style": Object {
                  "fontSize": "14pt",
                  "marginTop": 20,
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "2014",
                  },
                  "ref": null,
                  "rendered": "2014",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Directed by ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Directed by ",
                    undefined,
                  ],
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Written by ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Written by ",
                    undefined,
                  ],
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Cast: ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Cast: ",
                    undefined,
                  ],
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Language: ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Language: ",
                    undefined,
                  ],
                  "type": "li",
                },
                undefined,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Run Time: ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Run Time: ",
                    undefined,
                  ],
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "IMDB Rating: ",
                      undefined,
                      "/10",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "IMDB Rating: ",
                    undefined,
                    "/10",
                  ],
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Box Office: ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Box Office: ",
                    undefined,
                  ],
                  "type": "li",
                },
                undefined,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <a
                      href="https://www.imdb.com/title/tt2310332"
>
                      IMDB page
</a>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "IMDB page",
                      "href": "https://www.imdb.com/title/tt2310332",
                    },
                    "ref": null,
                    "rendered": "IMDB page",
                    "type": "a",
                  },
                  "type": "li",
                },
              ],
              "type": "ul",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Top",
                "onClick": [Function],
                "style": Object {
                  "marginBottom": 20,
                },
                "title": "Scroll to top of page",
              },
              "ref": null,
              "rendered": "Top",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders a spinner if isFetching 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilmDetail
    dispatchFetchFilmDetails={[Function]}
    filmDetails={
        Object {
            "Poster": "https://images-na.ssl-images-amazon.com/images/M/MV5BODAzMDgxMDc1MF5BMl5BanBnXkFtZTgwMTI0OTAzMjE@._V1_SX300.jpg",
            "Title": "The Hobbit: The Battle of the Five Armies",
            "Type": "movie",
            "Year": "2014",
            "imdbID": "tt2310332",
          }
    }
    imdbID="tt2310332"
    isFetching={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Unknown
        size={64}
/>,
      "style": Object {
        "display": "flex",
        "height": "100vh",
        "justifyContent": "center",
        "minHeight": "100vh",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "size": 64,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Unknown
          size={64}
/>,
        "style": Object {
          "display": "flex",
          "height": "100vh",
          "justifyContent": "center",
          "minHeight": "100vh",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "size": 64,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
